### PR TITLE
feat(dashboards): add widget edit

### PR DESCRIPTION
**Pull request by Kaskyi**
add yaml support
add tests

### CODE REVIEW

**Review by grankovsky — APPROVED** [review]

**Review by LevBD — APPROVED** [review]

File: packages/components/ace-editor/ace-editor-control/xm-ace-editor-control.ts [hunk 1]
```diff
@@ -69,56 +79,77 @@ type AceEditorValue = string | object;
...
 
-    public change(v: AceEditorValue): void {
-        if (this._config.mode === 'json') {
-            try {
-                v = JSON.parse(v as string);
-                this.error = false;
-            } catch (e) {
-                this.error = true;
-            }
+    public getMode(): string | null {
```
Comment (denisogolubev) [inline packages/components/ace-editor/ace-editor-control/xm-ace-editor-control.ts:129]: `object-to-yaml`, `yaml`, `json` these stirngs duplicated in component use enum.

Also this method could be simplified:
```ts
switch (this.config.mode) {
    case 'object-to-yaml':
    case 'yaml':
        return 'yaml';
    case 'object-to-json':
    case 'json':
        return 'json';
    default:
        return null;
}
```

File: packages/components/ace-editor/ace-editor-control/xm-ace-editor-control.ts [hunk 2]
```diff
@@ -69,56 +79,77 @@ type AceEditorValue = string | object;
...
     }
 
     @Input()
     public set value(value: AceEditorValue) {
-        if (typeof value === 'object') {
-            this._value = JSON.stringify(value, null, 2);
-        } else {
-            this._value = value;
+        try {
+            if (this.config.mode === 'object-to-yaml') {
```
Comment (denisogolubev) [inline packages/components/ace-editor/ace-editor-control/xm-ace-editor-control.ts:116]: it's better to use switch case here

**Comment by zverbeta** [comment]
@Kaskyi Did you consider create a new yaml control for this purpose, because this control doesn't work properly

**Comment by Kaskyi** [comment]
@zverbeta I don't know cases when it doesn't work.

**Review by Olensavchenko — APPROVED** [review]

File: packages/components/ace-editor/ace-editor-control/xm-ace-editor-control.ts (hunk 1 above)
Comment (Kaskyi) [reply packages/components/ace-editor/ace-editor-control/xm-ace-editor-control.ts:129]: i just like symmetry)

Comment (Kaskyi) [reply packages/components/ace-editor/ace-editor-control/xm-ace-editor-control.ts:129]: done

File: packages/components/ace-editor/ace-editor-control/xm-ace-editor-control.ts (hunk 2 above)
Comment (Kaskyi) [reply packages/components/ace-editor/ace-editor-control/xm-ace-editor-control.ts:116]: done

**Pull request merged**